src/Core/System/Snippet/SnippetFileHandler.php

```diff
@@ -7,28 +7,49 @@
 use Shopware\Storefront\Storefront;
 use Symfony\Component\Finder\Finder;
 
+/**
+ * @phpstan-type Snippets array<string, string|mixed>
+ */
 #[Package('discovery')]
 class SnippetFileHandler
 {
+    /**
+     * @return Snippets
+     */
     public function openJsonFile(string $path): array
     {
-        $json = json_decode(file_get_contents($path), true);
+        $fileContents = file_get_contents($path);
```

Comment:
would be nice, if you could refactor `file_get_contents` and similar native functions with the Symfony Filesystem component